content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/ContentEntityProvider.java

```diff
@@ -295,6 +304,50 @@ public boolean entityExists(String id) {
         }
 		return rv;
 	}
+
+	/**
+	 * return the original URL string based on Web Link resource content 
+	 * @param resource
+	 * @param contentType
+	 * @return
+	 */
+	private String getWebLinkResourceUrlString(ContentResource resource, String contentType) {
+		String urlString = null;
+		
+		// for Web Link resources
+		// return the original URL link, instead of access URL
+		if (contentType.equalsIgnoreCase(ResourceProperties.TYPE_URL))
+		{
+			try
+			{
+				byte[] content = resource.getContent();
```

Comment:
So what if this is a 1GB movie?  Did we just load the whole content into a byte array?